pkg/convert/tf.go

```diff
@@ -1141,14 +1141,77 @@ func convertTemplateWrapExpr(state *convertState,
 	return tokens
 }
 
+func convertTemplateJoinExpr(state *convertState, inBlock bool, scopes *scopes,
+	fullyQualifiedPath string, expr *hclsyntax.TemplateJoinExpr,
+) hclwrite.Tokens {
+	forExpr := expr.Tuple.(*hclsyntax.ForExpr)
+	if forExpr == nil {
```

Comment:
Won't this trigger a panic if the assertion fails? I feel like we need the two-value form, `forExpr, ok := ...`? I might be missing something though.

Reply:
I think checking nil works, but might as well use hte ok feature